### PR TITLE
[Sample] Add text classification app

### PR DESCRIPTION
**Pull request by gichan-jang**
Add text classification sample app with NNStreamer single-API

See : https://docs.tizen.org/application/native/guides/machine-learning/machine-learning-inference
See also : https://github.com/nnsuite/nnstreamer

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

### CODE REVIEW

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+namespace TextClassification
+{
+    /// <summary>
+    /// Handle the input text
+    /// </summary>
+    public class HandleText
+    {
+        const string TAG = "TextClassification.Models";
+
+        private static string ResourcePath = Tizen.Applications.Application.Current.DirectoryInfo.Resource;
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]:
```suggestion
        private static string ResourcePath { get => Tizen.Applications.Application.Current.DirectoryInfo.Resource; }
```

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+            return float_array;
+        }
+
+        /// <summary>
+        /// Load files and make hash table
+        /// </summary>
+        public void Init_model()
+        {
+            string label = ResourcePath + "models/labels.txt";
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]:
```suggestion
            string label = Path.Combine(ResourcePath, "models/labels.txt");
```
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netframework-4.8#System_IO_Path_Combine_System_String_System_String_

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                Log.Error(TAG, "Vocab file is not exist");
+            }
+        }
+
+        /// <summary>
+        /// Invoke text classification model and return result
+        /// </summary>
+        /// <param name="input"> float array which is converted from user input </param>
+        /// <returns> Probability of the paragraph being positive or negative </returns>
+        public float[] Invoke_mode(float[] input)
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]:
```suggestion
        public float[] Invoke_model(float[] input)
```

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        {
+            byte[] in_buffer = new byte[4 * 256];
+            byte[] out_buffer;
+            float[] output = new float[2];
+            TensorsInfo in_info;
+            TensorsInfo out_info;
+            TensorsData in_data;
+            TensorsData out_data;
+
+            string model_path = ResourcePath + "models/text_classification.tflite";
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]:
```suggestion
            string model_path = Path.Combine(ResourcePath, "models/text_classification.tflite");
```
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netframework-4.8#System_IO_Path_Combine_System_String_System_String_

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Event when H/W back button is clicked.
+        /// </summary>
+        /// <returns> true if exit is successful </returns>
+        protected override bool OnBackButtonPressed()
+        {
+            System.Environment.Exit(1);
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: Do not directly exit, 
If do not override this method, defaultly exit by UI frameworks

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+        /// <summary>
+        /// Event when "Get result" button is clicked.
+        /// </summary>
+        /// <param name="sender">Instance of the object which invokes event.</param>
+        /// <param name="args"> Event argument.</param>
+        public async void CreateClicked(object sender, EventArgs args)
+        {
+            Button button = (Button)sender;
+            InputLabel.Text = Comment;
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: If you set a name of Entry on XAML, you can refer text of entry 
``` xml
 <Entry x:Name="UserCommentEntry" Grid.Row="1" Grid.Column="0" Placeholder="Comment"
                 Keyboard="Default"
                 HorizontalOptions="FillAndExpand"
                 ClearButtonVisibility="WhileEditing"
                 BackgroundColor="GhostWhite"/>
```
``` c#
    InputLabel.Text = UserCommentEntry.Text;
```
And you anymore not need TextChanged handle (`UserComment`)

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            /// Convert user input to float array
+            input = TextModel.SplitStr(Comment);
+
+            if (string.IsNullOrEmpty(Comment))
+            {
+                Log.Warn(TAG, "User input is empty");
+                /// Check wrong Input parameter
+                await DisplayAlert("Error!", "Please enter your comment", "OK");
+            }
+            else if (Equals(button.Text, "Get Result"))
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: I don't know why need it?

Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: My intention is to distinguish the two button inputs when the pipeline API is added later.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs (hunk 1 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: Thank you. I will change it.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs (hunk 2 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: Thank you. I will change it.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs (hunk 3 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: My falut. Thank you!

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs (hunk 4 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: Thank you. I will change it.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            /// Initialize the model
+            TextModel.Init_model();
+            /// Convert user input to float array
+            input = TextModel.SplitStr(Comment);
+
+            if (string.IsNullOrEmpty(Comment))
+            {
+                Log.Warn(TAG, "User input is empty");
+                /// Check wrong Input parameter
+                await DisplayAlert("Error!", "Please enter your comment", "OK");
```
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: need to return in here

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs (hunk 5 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: I will remove this method. Thank you.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs (hunk 8 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: I will update it, Thank you

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs (hunk 4 above)
Comment (myroot) [inline Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: You need to write description about `text_classification.tflite`, where is come from, and how to generate it

Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Models/TextClassification.cs]: It is described briefly in README.md file.

File: Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs (hunk 6 above)
Comment (gichan-jang) [reply Mobile/MachineLearning/TextClassification/TextClassification/TextClassification/Views/UserInput.xaml.cs]: Updated. Thank you!

**Pull request merged**